plugins/circleci/src/util/time.ts

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2020 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import dayjs from 'dayjs';
```

Comment:
Wondering if this could be useful in `core` possibly? Also the `cost insights` plugin introduced using `dayjs`. I'm thinking it would be a good idea to move away from using `moment` as per https://momentjs.com/docs/#/-project-status/. If so, maybe an ADR?

Reply:
@andrewthauer yep - I think it would be awesome if you can put together an ADR for this. I know `moment` is going :dead:, and on my previous squad in Spotify we settled on `luxon` also from the `moment` team https://moment.github.io/luxon. I do remember also from the conversations that we had that  `dayjs` did pop up, so I'd be interested in seeing what you're thoughts are between the two

Reply:
@backstage/silver-lining you had some opinions here right? I seem to remember you picking one of them first, but then switching to the other because of reasons

Reply:
(And yes, absolute +1 for an ADR)

Reply:
I've never used `luxon`, and this was my first attempt using `dayjs`. In the past I've used `date-fns` which I liked quite a bit because it takes an FP approach which allows for composition chains. Dayjs seems familiar to `moment` and was decently enjoyable and easy to use. Seems to cover most use cases that would be needed.

Reply:
Here a good comparison I found - https://medium.com/swlh/best-moment-js-alternatives-5dfa6861a1eb

Reply:
@freben Yeah we are in search of a `moment` replacement - we've been trying out `dayjs` but @brendasukh found it difficult to work with for some of our use cases. `luxon` is the next obvious choice. Our needs are date addition/subtraction, including ISO 8601 period support (e.g. `P1M`, `P30D`), and date truncation (e.g. start of quarter).

Reply:
@timbonicus - Have you looked at `date-fns`? It's another very popular option which should cover all of these cases.

Reply:
I think this PR is pretty good to go, shall we move out the ADR work seperately as not to block this.

Reply:
I can create an ADR starting point and continue the discussion there. Or would an RFC to start with make more sense?

Reply:
Maybe and RFC and we can gather other community members input too in one place! I'll go ahead and merge this now.

Reply:
RFC added - #3401

Reply:
We haven't looked at `date-fns` but will do so - thanks for the pointer and RFC!